### ISSUE TITLE
Migrate to downstream tasks

### ISSUE THREAD

**Issue by OPilgrim** [princeton-nlp/TRIME#2]
Dear author, I want to know what I should do if I want to apply Trime-long to a specific downstream task, such as the relation extraction of a Long text, and use the "BART" of Hugging face to realize the sequence generation. I have a few ideas: 1. Find a way to replace BART's loss calculation with trime_long_loss.py, and use BART's pre-trained checkppoint to fine-tune directly on downstream tasks. 2. Pre-train Bart+Trime_Long in other datasets before the previous step. I am not sure whether the second step is needed or even whether the idea of such loss migration is correct. I hope you can give me some suggestions, thank you very much.

**Comment by a3616001** [comment]
Hi, thanks for your interests! Do you manage to formalize the downstream task as a generation problem? If so, I think it is doable and it'd be interesting to see how it performs on long text!

I would suggest to start from (1), as (2) might be very expensive. Also, there is something between (1) and (2): you can take a pre-trained checkpoint; continue pre-training with long-term memory on an unlabeled corpus; and then fine-tune the model on downstream tasks.

Hope this helps!
